cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -992,6 +1036,26 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 
 		return machine, nil
 	}()
+	if tryingCurrentSystem {
+		// end of the line for a recovery system we are only trying out,
+		// this branch always ends with a reboot (or a panic)
+		outcome := boot.TryRecoverySystemOutcomeFailure
+		if err == nil && !machine.degraded() {
+			outcome = boot.TryRecoverySystemOutcomeSuccess
+		}
+		finalizeErr := finalizeTryRecoverySystemAndReboot(outcome)
+		// unreachable unless in tests, as finalize issues a reboot,
+		// waits for it and panics if none happened
+		status := "failed"
+		if outcome == boot.TryRecoverySystemOutcomeSuccess {
+			status = "successful"
+		}
+		if finalizeErr != nil {
+			err = finalizeErr
+		}
+		panic(fmt.Errorf("%v tried recovery system %q: %v", status, mst.recoverySystem, err))
```

Comment:
successful/failed tried reads a bit oddly ?

Reply:
Yeah, I tried to make it too fancy. Dropped this now. Also made sure that relevant errors are actually logged, so that at least have some trace of what failed.